pypose/optim/optimizer.py

```diff
@@ -451,17 +434,28 @@ def step(self, input, target=None, weight=None):
             <https://github.com/pypose/pypose/tree/main/examples/module/pgo>`_.    
         '''
         for pg in self.param_groups:
+            weight = self.weight if weight is None else weight
             R = self.model(input, target, weight)
             J = modjac(self.model, input=(input, target, weight), **self.jackwargs)
             R, J = self.corrector(R = R, J = J)
             self.last = self.loss = self.loss if hasattr(self, 'loss') \
                                     else self.model.loss(input, target, weight)
-            A, self.reject_count = J.T @ J, 0
+            # Let Nr: number of residuals, Dr: dimension of each residual, Np: number of model parameters
+            # Now, the shape of R: [Nr, Dr], J: [Nr*Dr, Np], weight: [Nr, Dr, Dr]
```

Comment:
This is not correct, `R` can be any higher dimension, depending on the model defined by the users. I will and a test example to the test folder.